sockets.go

```diff
@@ -157,6 +157,49 @@ type JSONConnection struct {
 	Receiver reflect.Value
 }
 
+// Binary Connection connects a websocket message connection to a []byte
+// channel.
+type BinaryConnection struct {
```

Comment:
This should be renamed to e.g. `ByteSliceConnection`. `BinaryConnection` suggests a `[]bool`, but we have a `[]byte`